### PR TITLE
refactor(framework):clean up function analyzer #780 

### LINKED ISSUE #780: clean up function analyzer
extract the repeated regexp
image

https://sonarcloud.io/project/issues?resolved=false&id=Idrinth_api-bench

### PR DESCRIPTION
**Pull request by akhil0203**
# The Pull Request is ready

- [x] fixes #780 
- [x] all actions are passing

## Overview

RegExp.exec() method is used instead of string.match() as suggested in sonar cloud analysis
multiple module-global constants are created for repeated RegExp.

## Review points

creation of multiple module-global constants

## Framework

- [x] default behaviour did not change

### CODE REVIEW

**Comment by Idrinth** [comment]
@akhil0203  please restore the eslint ignores properly, then everything should pass again, otherwise good work!

**Review by Idrinth — CHANGES_REQUESTED** [review]
- rename regexp for what the do, not where they are used
- restore eslint-disable-next-line to make eslint pass

**Comment by Idrinth** [comment]
Hi @akhil0203 ,
This looks good, just can't be merged right now due to conflicts. Please merge the-one into this.

File: framework/src/helper/function-analyzer.ts [hunk 1]
```diff
@@ -24,9 +24,15 @@
...
-
-// eslint-disable-next-line complexity
+const commentAndAssignmentRegExp=/\/\*.+\*\/.+=.+/u;
+const commentRegExp=/\/\*.+\*\/.+/u;
+const assignmentRegExp=/.+=.+/u;
+const commentOrAssignmentAtEndRegExp =/\/\*.+\*\/|=.+$/gu;
+const whitespaceRegExp =/\s*/gu;
+const beginningAssignmentRegExp=/^.+=/u;
+const leadingOrTrailingWhitespaceRegExp=/^\s*|\s*$/gu;
+const commentBlockOrWhitespaceRegExp=/^.*\/\*|\*\/.+$/gu;
```
[the remark this answers is not in the document]
Comment (Idrinth) [reply framework/src/helper/function-analyzer.ts:34]: @coderabbitai unicode matching is required by default in this project.

**Comment by akhil0203** [comment]
> * rename regexp for what the do, not where they are used
> * restore eslint-disable-next-line to make eslint pass



> Hi @akhil0203 , This looks good, just can't be merged right now due to conflicts. Please merge the-one into this.

hi @Idrinth its done please review it.Thanks

**Pull request merged**